lib/loggingService/LogEntry.js

```diff
@@ -247,17 +247,11 @@ LogEntry.prototype.addDetails = function (args) {
 
 
 LogEntry.prototype.addData = function (data) {
-	data = serializeAny(data);
-
-	if (this.data) {
-		for (var key in data) {
-			if (data.hasOwnProperty(key)) {
-				this.data[key] = data[key];
-			}
-		}
-	} else {
-		this.data = data;
+	if (!this.data) {
+		this.data = [];
```

Comment:
We push data in an array of objects, and the value is set as an attribute of that object. Data should be anything you want, including undefined.